x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/ingest/GeoTileProcessor.java

```diff
@@ -0,0 +1,421 @@
+                    } else {
+                        yield H3.stringToH3(H3.h3ToParent(h3));
+                    }
+                }
+                default -> throw new IllegalArgumentException("Unknown H3 function call [" + parts[0] + "]");
+            };
+        }
+
+        private static void assertParams(String function, int len, int expected) {
+            if (len < expected) {
+                throw new IllegalArgumentException(
+                    "Function H3." + function + " requires " + expected + " parameters but " + len + " found"
+                );
+            }
+        }
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    String field(String key) {
+        return config.field(key);
+    }
+
+    GeometryParserFormat targetFormat() {
+        return targetFormat;
+    }
+
+    TileFieldType tileType() {
+        return tileFieldType;
+    }
+
+    public static final class Factory implements Processor.Factory {
+
+        public GeoTileProcessor create(
+            Map<String, Processor.Factory> registry,
+            String processorTag,
+            String description,
+            Map<String, Object> config
+        ) {
+            String field = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "field");
+            String targetField = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "target_field", field);
+            String parentField = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "parent_field", "");
+            String childrenField = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "children_field", "");
+            String nonChildrenField = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "non_children_field", "");
+            String depthField = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "depth_field", "");
+            boolean ignoreMissing = ConfigurationUtils.readBooleanProperty(TYPE, processorTag, config, "ignore_missing", false);
+            TileFieldType tileFieldType = TileFieldType.parse(
+                ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "tile_type")
+            );
+            GeometryParserFormat targetFormat = getTargetFormat(processorTag, config);
+            FieldConfig fields = new FieldConfig(field, targetField, parentField, childrenField, nonChildrenField, depthField);
+            return new GeoTileProcessor(processorTag, description, fields, ignoreMissing, targetFormat, tileFieldType);
+        }
+
+        private GeometryParserFormat getTargetFormat(String processorTag, Map<String, Object> config) {
+            String propertyName = "target_format";
+            String targetFormat = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, propertyName, "WKT");
+            return switch (targetFormat.toLowerCase(Locale.ROOT).trim()) {
```

Comment:
In general our default format is normally gejson

Reply:
OK. Done